### ISSUE TITLE
Incorrect narrowing for class pattern when `Callable` is used

### ISSUE THREAD

**Issue by randolf-scholz** [microsoft/pyright#6648]
Possibly even bug, maybe related to #6600 ?

```python
from collections.abc import Callable
from typing import TypeVar

T = TypeVar("T")


def foo(obj: T | Callable[..., T]) -> None:
    match obj:
        case Callable() as func:
            func()
```

I guess the issue is that if `T` is callable, then its signature might be incompatible? But for completely unknown `T`, I'd assume that `func` should be inferred as the widest possible type `Callable[..., Any]`.

**Comment by erictraut** [comment]
Yes, this is a bug. The current type narrowing logic for class pattern matching doesn't properly handle `Callable`.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
